util/process.go

```diff
@@ -42,3 +42,21 @@ func AwaitSignal(ctx context.Context, doneFn func()) {
 		glog.Infof("AwaitSignal canceled: %v", ctx.Err())
 	}
 }
+
+// AwaitContext waits for context done, then runs the given function.
```

Comment:
I'd be tempted to keep this as an unexported function in the `serverutil` package it's called from...

Reply:
fixed